### PR TITLE
fix redux

### PR DESCRIPTION
**Pull request by Wyfy0107**
Please review this thoroughly since it introduces lots of changes.
🐛 in redux has been fixed with magical `@ts-ignore` but I think our code is fine but rather it's redux's weird behavior sometimes

### CODE REVIEW

File: src/components/NavBar/index.tsx [hunk 1]
```diff
@@ -23,35 +23,37 @@ const NavBar = () => {
...
-            <LinkContainer to='/profile'>
-              <NavDropdown.Item>Profile</NavDropdown.Item>
+          {!userInfo && (
+            <LinkContainer to="/register">
+              <Nav.Link>
+                <i className="register"></i>Register
+              </Nav.Link>
             </LinkContainer>
-            <LinkContainer to={`/userpage/${userInfo.id}`}>
-              <NavDropdown.Item>Home</NavDropdown.Item>
```
Comment (Tsaikkari) [inline src/components/NavBar/index.tsx:42]: No link to user homepage? I didn't see a link in any pages in figma, so I added it to the navbar.

File: src/pages/Profile/index.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          <div className="work-experience">
+            <h3>Work Experience</h3>
+            <p className="work-experience__title">Integrify</p>
+            <p className="work-experience__period">2 Years</p>
+          </div>
+          <h3>Level</h3>
+          <p>Junior Developer</p>
+          <Button variant="success">Chat</Button>
+        </Col>
+      </Row>
```
Comment (Tsaikkari) [inline src/pages/Profile/index.tsx:38]: Putting everything inside one column and row is somewhat original. ;)

File: src/redux/saga/employer.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { registerEmployerSuccess } from '../../redux/actions/employer';
+
+function* registerEmployerSaga(credentials: Credential) {
+  try {
+    const req = yield axios.post('/employer', { credentials });
+    yield put(registerEmployerSuccess(req.data));
+  } catch (error) {
+    console.log(error);
+  }
+}
```
Comment (Tsaikkari) [inline src/redux/saga/employer.ts:13]: If I hit the https://tindev-dev-deploy.herokuapp.com/employer here I'll get the cors error, so maybe we should use cors in the backend?

File: src/redux/store.ts [hunk 4]
```diff
@@ -16,30 +16,27 @@ export default function makeStore(initialState = initState) {
...
   if (process.env.NODE_ENV === 'development') {
     if ((window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
       composeEnhancer = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
     }
   }
 
   const store = createStore(
-    rootReducers,
+    rootReducer(),
+    //@ts-ignore
```
Comment (Tsaikkari) [inline src/redux/store.ts:27]: Just love the magic of ts-ignore! :D <3

**Review by Tsaikkari — COMMENTED** [review]
It works now somewhat better. :)

File: src/redux/saga/employer.ts (hunk 3 above)
Comment (Wyfy0107) [reply src/redux/saga/employer.ts:13]: Can you create an issue on the backend for this ?

File: src/pages/Profile/index.tsx (hunk 2 above)
Comment (Wyfy0107) [reply src/pages/Profile/index.tsx:38]: I checked the develop branch and see this is similar to it. Which branch are you referring to ?

**Pull request merged**